pkg/runtime/reconciler.go

```diff
@@ -604,27 +636,25 @@ func (r *resourceReconciler) patchResourceMetadataAndSpec(
 	}
 
 	rlog.Enter("kc.Patch (metadata + spec)")
-	// Save a copy of the latest object, to reset 'Status' after performing
-	// the kc.Patch() operation
-	latestCopy := latest.DeepCopy()
-	err = r.kc.Patch(
-		ctx,
-		latest.RuntimeObject(),
-		client.MergeFrom(desired.DeepCopy().RuntimeObject()),
-	)
-	// Reset the status of latest object after patching.
-	latest.SetStatus(latestCopy)
-	rlog.Exit("kc.Patch (metadata + spec)", err)
-
-	if err != nil {
-		return err
+	dobj := desired.DeepCopy().RuntimeObject()
+	lobj := latest.DeepCopy().RuntimeObject()
+	patch := client.MergeFrom(dobj)
```

Comment:
We have tried doing something like this, I think. When we patch spec or metadata, it updates the `resourceVersion` and any of our existing references cannot be used for patching.

Reply:
@RedbackThomson I was under the impression that would only impact us if we were using optimistic locking option:

https://github.com/kubernetes-sigs/controller-runtime/blob/b704f447ea7c8f7059c6665143a4aa1f6da28328/pkg/client/patch.go#L81-L84

That said, I'm happy to revert this part of the patch if you're concerned about it. All e2e tests were passing in the ecr controller local build-kind tests I ran.